src/Microsoft.Azure.SignalR.Common/Endpoints/AccessKeyManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class AccessKeySynchronizer : IAccessKeySynchronizer, IDisposable
+    {
+        private readonly ConcurrentDictionary<ServiceEndpoint, object> _endpoints = new ConcurrentDictionary<ServiceEndpoint, object>(new ServiceEndpointsComparer());
+
+        private readonly TimerAwaitable _timer = new TimerAwaitable(TimeSpan.Zero, TimeSpan.FromMinutes(5));
+
+        private readonly IServerNameProvider _provider;
+
+        private readonly ILoggerFactory _factory;
+
+        public AccessKeySynchronizer(
+            IServerNameProvider nameProvider,
+            ILoggerFactory loggerFactory,
+            bool test = false
+            )
+        {
+            if (!test)
+            {
+                _ = UpdateAsync();
+            }
+
+            _provider = nameProvider;
+            _factory = loggerFactory;
+        }
+
+        public bool AddServiceEndpoint(ServiceEndpoint endpoint)
+        {
+            if (endpoint.AccessKey is AadAccessKey aadKey)
+            {
+                _ = aadKey.UpdateAccessKeyAsync(_provider, _factory);
+            }
+            return _endpoints.TryAdd(endpoint, null);
+        }
+
+        public void AddServiceEndpoints(IEnumerable<ServiceEndpoint> endpoints)
+        {
+            foreach (var endpoint in endpoints)
+            {
+                _endpoints.TryAdd(endpoint, null);
+            }
+        }
+
+        public void Dispose() => _timer.Stop();
```

Comment:
>Stop [](start = 40, length = 4)

why change from `((IDisposable)_timer).Dispose();` to `stop`

Reply:
@vwxyzh